virttest/guest_agent.py

```diff
@@ -509,7 +509,10 @@ def shutdown(self, mode=SHUTDOWN_MODE_POWERDOWN):
         if mode in [self.SHUTDOWN_MODE_POWERDOWN, self.SHUTDOWN_MODE_REBOOT,
                     self.SHUTDOWN_MODE_HALT]:
             args = {"mode": mode}
-        self.cmd(cmd=cmd, args=args, success_resp=False)
+        try:
+            self.cmd(cmd=cmd, args=args, success_resp=True)
+        except VAgentCmdError:
+            raise
```

Comment:
The above approach has the same meaning as `self.cmd(cmd=cmd, args=args, success_resp=True)`, the problem is there are several types of error but only some of them would not be allowed.